mcs/class/corlib/Test/System.Reflection.Emit/DynamicMethodTest.cs

```diff
@@ -475,6 +475,9 @@ public void GetCurrentMethodWorksWithDynamicMethods ()
 	        DynamicMethod dm = new DynamicMethod("Foo", typeof(object), null);
 	        ILGenerator ilgen = dm.GetILGenerator();
 	        ilgen.Emit(OpCodes.Call, typeof(MethodBase).GetMethod("GetCurrentMethod"));
+		// Call twice and pop to avoid tailcall optimization.
+	        ilgen.Emit(OpCodes.Call, typeof(MethodBase).GetMethod("GetCurrentMethod"));
```

Comment:
If it works, this might be better of call to noinline NoTailcall like the others.
As well, it might be reasonable for JIT to special case GetCurrentMethod -- it already does sometimes.